schemafy_lib/src/generator.rs

```diff
@@ -33,22 +39,21 @@ impl<'a, 'b> Generator<'a, 'b> {
         GeneratorBuilder::default()
     }
 
-    pub fn generate_with_schema<'c>(&'c self) -> (proc_macro2::TokenStream, Schema) {
-        let json = if let Some(input_file) = self.input_file {
-            let input_file = if input_file.is_relative() {
-                let crate_root = get_crate_root().unwrap();
-                crate_root.join(input_file)
-            } else {
-                PathBuf::from(input_file)
-            };
+    pub fn generate_with_schema(&self) -> (proc_macro2::TokenStream, Schema) {
+        let json = match self.input {
+            Input::File(input_file) => {
+                let input_file = if input_file.is_relative() {
+                    let crate_root = get_crate_root().unwrap();
+                    crate_root.join(input_file)
+                } else {
+                    PathBuf::from(input_file)
+                };
 
-            std::fs::read_to_string(&input_file).unwrap_or_else(|err| {
-                panic!("Unable to read `{}`: {}", input_file.to_string_lossy(), err)
-            })
-        } else if let Some(input_json) = &self.input_json {
-            input_json.clone()
-        } else {
-            panic!("Either input file or input json should be specified");
+                std::fs::read_to_string(&input_file).unwrap_or_else(|err| {
+                    panic!("Unable to read `{}`: {}", input_file.to_string_lossy(), err)
+                })
+            }
+            Input::Json(input_json) => input_json.to_owned(),
```

Comment:
nit: you are just using the owned value to deserialize using a reference to it. Would it make sense to just deserialize within this match block to avoid this? Doesn't really matter, fine to keep as is as the code is slightly cleaner

(this was the point of only passing in a reference as input)